### PR TITLE
PARQUET-2489: Guidance on Parquet-Java releases

### PR DESCRIPTION
**Pull request by emkornfield**

### CODE REVIEW

File: content/en/docs/Contribution Guidelines/modules.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ description: >
 
 The [parquet-format](https://github.com/apache/parquet-format) project contains format specifications and Thrift definitions of metadata required to properly read Parquet files.
 
-The [parquet-mr](https://github.com/apache/parquet-mr) project contains multiple sub-modules, which implement the core components of reading and writing a nested, column-oriented data stream, map this core onto the parquet format, and provide Hadoop Input/Output Formats, Pig loaders, and other Java-based utilities for interacting with Parquet.
+The [parquet-java](https://github.com/apache/parquet-java) project contains multiple sub-modules, which implement the core components of reading and writing a nested, column-oriented data stream, map this core onto the parquet format, and provide Hadoop Input/Output Formats, Pig loaders, and other Java-based utilities for interacting with Parquet.
```
Comment (alamb) [inline content/en/docs/Contribution Guidelines/modules.md]: Here is a suggestion to clean up the language to more closely mirror the structure of the cpp :

```suggestion
The [parquet-java](https://github.com/apache/parquet-java) project is a Java library to read and write Parquet files. It consists of multiple sub-modules, which implement the core components of reading and writing a nested, column-oriented data stream, to and from the Parquet format, along with Hadoop Input/Output Formats, Pig loaders, and other Java-based utilities for interacting with Parquet.
```

File: content/en/docs/Contribution Guidelines/releasing.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ title: "Releasing Parquet"
 linkTitle: "Releasing Parquet"
 weight: 4
 description: >
-  How to release Parquet
+  How to release Parquet-Java
```
Comment (jorisvandenbossche) [inline content/en/docs/Contribution Guidelines/releasing.md:6]: Maybe you can update the `title` to include "Java" as well? (as it's the title that is most visible)

File: content/en/docs/Contribution Guidelines/releasing.md [hunk 3]
```diff
@@ -173,3 +173,18 @@ Update the downloads page on parquet.apache.org. Instructions for updating the s
     Java artifacts are available from Maven Central.
 
     Thanks to everyone for contributing!
+
+#### 6\. Update parquet-format with feature enablement guidance
+
+The recommendations for other feature enablment is generally tied to releases of
```
Comment (jorisvandenbossche) [inline content/en/docs/Contribution Guidelines/releasing.md]:
```suggestion
The recommendations for other feature enablement is generally tied to releases of
```

File: content/en/docs/Contribution Guidelines/releasing.md (hunk 2 above)
Comment (emkornfield) [reply content/en/docs/Contribution Guidelines/releasing.md:6]: done.

**Review by xhochy — APPROVED** [review]
Sounds good.

File: content/en/docs/Contribution Guidelines/releasing.md [hunk 4]
```diff
@@ -1,9 +1,9 @@
 ---
-title: "Releasing Parquet"
-linkTitle: "Releasing Parquet"
+title: "Releasing Parquet-Java"
+linkTitle: "Releasing Parquet-Java"
 weight: 4
 description: >
-  How to release Parquet
+  How to release Parquet-Java
```
Comment (wgtmac) [inline content/en/docs/Contribution Guidelines/releasing.md:6]: Releasing parquet-format is pretty much the same process as releasing parquet-java. Perhaps we can add a line to explain this?

File: content/en/docs/Contribution Guidelines/releasing.md [hunk 5]
```diff
@@ -173,3 +173,18 @@ Update the downloads page on parquet.apache.org. Instructions for updating the s
...
+parquet-java (details are in the parquet-format repo).  As releases are made the
+specification should be updated to indicate the recommended dates for when a new
+feature may be enabled.
+
+
+### Release Cadence
+
+Provided enough volunteers are available the Parquet community aims to have
+releases on a quarterly basis (Targets months are January, April, July and
+October). If a new major version is necessary it will be targetted for the
```
Comment (wgtmac) [inline content/en/docs/Contribution Guidelines/releasing.md:191]: I'm fine to release a major version in every October.

However, as I have mentioned in https://github.com/apache/parquet-format/pull/258#discussion_r1660557601, a lot of deprecations are marked to be removed in 2.0.0 like https://github.com/apache/parquet-java/blob/e6d7728267691a1ff433358b6c60d0cae38b5e36/parquet-common/src/main/java/org/apache/parquet/Closeables.java#L29.

Perhaps this October we should just release 1.15.0 to keep the API compatibility. We have enough time to prepare for 2.0.0 next year.

cc @gszadovszky @Fokko @julienledem

Comment (gszadovszky) [reply content/en/docs/Contribution Guidelines/releasing.md:191]: TBH, I am not a big fun of doing a major release only to remove deprecated parts. But let's discuss the 2.0.0 content and release schedule on the dev list.

Comment (Fokko) [reply content/en/docs/Contribution Guidelines/releasing.md:191]: I do think it is time for a new major release to remove a lot of legacy (for example the removal of OriginalType https://github.com/apache/parquet-java/pull/1194). However, we don't want to be too aggressive, so this is a balance we have to find. I do think adding a 1.15.0 release might smooth out things, for example, by raising the lower bound of Hadoop step by step (>2.9 in 1.15 in https://github.com/apache/parquet-java/pull/2944, and then we can move to >3.3 at some point in 2.0?)

Comment (Fokko) [reply content/en/docs/Contribution Guidelines/releasing.md:191]: We could communicate the dates to the public by setting dates on the milestones: https://github.com/apache/parquet-java/milestones

Comment (emkornfield) [reply content/en/docs/Contribution Guidelines/releasing.md:191]: Dates on milestones sounds good to me.  I can start a discussion thread on October vs the new year, and we can always update accordingly.  I don't have a strong preference.

**Review by Fokko — APPROVED** [review]

**Comment by Fokko** [comment]
@emkornfield Looks like there are conflicts

**Comment by emkornfield** [comment]
> @emkornfield Looks like there are conflicts

Yes, I will try to resolve them this week

File: content/en/docs/Contribution Guidelines/releasing.md (hunk 4 above)
Comment (emkornfield) [reply content/en/docs/Contribution Guidelines/releasing.md:6]: Added below.

File: content/en/docs/Contribution Guidelines/releasing.md (hunk 5 above)
Comment (emkornfield) [reply content/en/docs/Contribution Guidelines/releasing.md:191]: Rebased.

**Review by wgtmac — APPROVED** [review]

**Comment by Fokko** [comment]
Thanks for adding this @emkornfield I think this helps set expectations for the community. Thanks @emkornfield @alamb @jorisvandenbossche and @xhochy for the review 👍  I'll go ahead and merge this since it has been open for a while and there seems to be consensus 👍

**Pull request merged**